### PR TITLE
Add option for specifying locale as an option

### PR DESCRIPTION
**Pull request by tmorehouse**
Adds option to allow locale to be specified as a option.  Assumes that the required locale files have been pre-loaded.

First attempts to find local as specified and uses it if found.
Then tries checking if local was specified with `_` insterad of `-` (translates `_` to `-`)
And if that is not found it tries for a short language only locale (ie.e `en` instead of `en-US`)
And if that is not found, then the defaults are left as is.

Example:

`<table data-toggle="table" data-locale='en_ZZ"></table>`

- First tries to extend with `en_ZZ`, but cant find it,
- Then tries `en-ZZ`, but cant find it
- Then tries `en`, and finds it (it is an alias of `en-US`)

### CODE REVIEW

**Comment by djhvscf** [comment]
Thanks. I will check/test it

**Comment by tmorehouse** [comment]
I have this running in my production test environment now.  works great.  The only think I have changed (compared to the above pull request) is set the default locale to `en-US`, rather than `undefined`, in the `BootStapTable.DEFAULTS`  So that it always defaults to `en-US` regardless of the order the language files are loaded.

**Comment by djhvscf** [comment]
Can you provide a small example, please?

**Comment by tmorehouse** [comment]
Here are a few examples:

	<link rel="stylesheet" href="lib/bootstrap-table.min.css">
	<script src="lib/jqurey.min.js"></script>
	<script src="lib/bootstrap-table.js"></script>
	<script src="lib/bootstrap-table-locale-all.js"></script>

	
	<table id="foo" 
		data-toggle="table" 
		data-locale="fr-FR"
		data-search="true"
		data-pagination="
		data-url="something.json">
		<thead>
			<tr>
				<th data-field="col1">Col 1</th>
				<th data-field="col2">Col 2</th>
				<th data-field="col3">Col 3</th>
			</tr>
		</thead>
	</table>

	
	<table id="bar">
		<thead>
			<tr>
				<th data-field="col1">Col 1</th>
				<th data-field="col2">Col 2</th>
				<th data-field="col3">Col 3</th>
			</tr>
		</thead>
	</table>

	
	<table id="baz" 
		data-toggle="table" 
		data-locale="zz-ZZ"
		data-search="true" 
		data-pagination="
		data-url="something.json">
		<thead>
			<tr>
				<th data-field="col1">Col 1</th>
				<th data-field="col2">Col 2</th>
				<th data-field="col3">Col 3</th>
			</tr>
		</thead>
	</table>

	
	<table id="fiz">
		<thead>
			<tr>
				<th data-field="col1">Col 1</th>
				<th data-field="col2">Col 2</th>
				<th data-field="col3">Col 3</th>
			</tr>
		</thead>
	</table>

	<script>
	$('#bar').bootstrapTable({
		// Notice "_". Will try for `de_DE, but if that doesn't exist tries 'de-DE'
		// and if not found tries 'de' and if not found, then use s the last language
		// from lib/bootstrap-table-locale-all.js, or what ever language is in
		// BootstrapTable.DEFAULTS.locale
		locale: 'de_DE',
		url: 'something.json',
		pagination: true
	});
	$('#fiz').bootstrapTable({
		// Will try first for en-ZZ, but if that doesn't exist tries 'en', 
		// and if not found uses default/last loaded language from
		// lib/bootstrap-table-locale-all.js, or what ever language is in
		// BootstrapTable.DEFAULTS.locale
		locale: 'en-ZZ', 
		url: 'something.json',
		pagination: true
	});
	</script>

And as usually, once can combine data attributes with manually instantiated tables (i.e. set locale and URL as data atributes, but passs other options via jQuery)

**Comment by tmorehouse** [comment]
This could be written as a small extension to bootstrapTable, rather than part of the core, but I beleive this feature should be part of the core (as it is only a few lines of additional code)

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -450,6 +452,24 @@
         this.initServer();
     };
 
+    BootstrapTable.prototype.initLocale = function () {
+		if (this.options.locale) {
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: Please use 4 spaces instead of tab.

**Comment by wenzhixin** [comment]
Thanks for your PR @tmorehouse , it sounds great! Can you help to add the option to documentation: https://github.com/wenzhixin/bootstrap-table/blob/master/docs/_i18n/en/documentation/table-options.md.

**Comment by tmorehouse** [comment]
Replaced the tabs with spaces, and I have added some documenttion. I hate writing documentation ;-) hehe

File: docs/_i18n/en/documentation/table-options.md [hunk 2]
```diff
@@ -489,5 +489,24 @@ The table options are defined in `jQuery.fn.bootstrapTable.defaults`.
         Support all custom attributes.
         </td>
     </tr>
-    </tbody>
+     <tr>
+        <td>locale</td>
+        <td>data-local</td>
```
Comment (djhvscf) [inline docs/_i18n/en/documentation/table-options.md]: Here should be data-locale, right?

Comment (tmorehouse) [inline docs/_i18n/en/documentation/table-options.md]: That is correct. me and my ten left fingers.  Will fix.

**Comment by tmorehouse** [comment]
Includes a fix for #1039

**Pull request merged**